src/System.Net.Security/src/System/Net/NegotiateStreamPal.Windows.cs

```diff
@@ -0,0 +1,467 @@
+                        securityContext,
+                        out token);
+                    if (winStatus != Interop.SecurityStatus.OK)
+                    {
+                        throw new Win32Exception((int)winStatus);
+                    }
+                    string authtype = context.ProtocolName;
+
+                    // TODO #5241:
+                    // The following call was also specifying WindowsAccountType.Normal, true.
+                    // WindowsIdentity.IsAuthenticated is no longer supported in CoreFX.
+                    result = new WindowsIdentity(token.DangerousGetHandle(), authtype);
+                    return result;
+                }
+                catch (SecurityException)
+                {
+                    // Ignore and construct generic Identity if failed due to security problem.
+                }
+                finally
+                {
+                    if (token != null)
+                    {
+                        token.Dispose();
+                    }
+                }
+            }
+
+            // On the client we don't have access to the remote side identity.
+            result = new GenericIdentity(name, protocol);
+            return result;
+        }
+
+        internal static string QueryContextAssociatedName(SafeDeleteContext securityContext)
+        {
+            return SSPIWrapper.QueryContextAttributes(GlobalSSPI.SSPIAuth, securityContext, Interop.SspiCli.ContextAttribute.Names) as string;
+        }
+
+        internal static string QueryContextAuthenticationPackage(SafeDeleteContext securityContext)
+        {
+            var negotiationInfoClass = SSPIWrapper.QueryContextAttributes(GlobalSSPI.SSPIAuth, securityContext, Interop.SspiCli.ContextAttribute.NegotiationInfo) as NegotiationInfoClass;
+            if (negotiationInfoClass != null)
+            {
+                return negotiationInfoClass.AuthenticationPackage;
+            }
+            return null;
+        }
+
+        internal static int QueryMaxTokenSize(string package)
+        {
+            return SSPIWrapper.GetVerifyPackageInfo(GlobalSSPI.SSPIAuth, package, true).MaxToken;
+        }
+
+        internal static string QueryContextClientSpecifiedSpn(SafeDeleteContext securityContext)
+        {
+            return SSPIWrapper.QueryContextAttributes(GlobalSSPI.SSPIAuth, securityContext, Interop.SspiCli.ContextAttribute.ClientSpecifiedSpn) as string;
+        }
+
+        internal static SafeFreeCredentials AcquireDefaultCredential(string package, bool isServer)
+        {
+            return SSPIWrapper.AcquireDefaultCredential(
+                GlobalSSPI.SSPIAuth,
+                package,
+                (isServer ? Interop.SspiCli.CredentialUse.Inbound : Interop.SspiCli.CredentialUse.Outbound));
+        }
+
+        internal static SafeFreeCredentials AcquireCredentialsHandle(string package, bool isServer, NetworkCredential credential)
+        {
+            unsafe
```

Comment:
Nit: rather than wrapping the entire body, this could just be in the signature line